### PR TITLE
Adding structuredLogging check and fix inconsistencies

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
After this PR the structuredLogging is enforced by govet check in the CI.

### CODE REVIEW

**Comment by lindy65** [comment]
Hi @jespino - can you help with QA test steps for this PR please?  Also, is there a Jira ticket for it?

**Comment by jespino** [comment]
@lindy65 there is no Jira ticket. and I added the label for QA because I didn't want to go directly with "no test required". Here we are only changing how the logs are generated, it is not changing any regular-user facing thing.

File: api4/system_test.go [hunk 1]
```diff
@@ -242,7 +242,8 @@ func TestGetLogs(t *testing.T) {
 	Client := th.Client
 
 	for i := 0; i < 20; i++ {
-		mlog.Info(fmt.Sprint(i))
+		logMessage := fmt.Sprint(i)
+		mlog.Info(logMessage)
```
Comment (hanzei) [inline api4/system_test.go]: It's is a bit artificial to create a tmp variable for this. What do you think about:
```suggestion
		mlog.Info(strconv.Itoa(i))
```

File: app/ldap.go [hunk 2]
```diff
@@ -19,7 +18,8 @@ func (a *App) SyncLdap() {
 			if ldapI := a.Ldap; ldapI != nil {
 				ldapI.StartSynchronizeJob(false)
 			} else {
-				mlog.Error(fmt.Sprintf("%v", model.NewAppError("SyncLdap", "ent.ldap.disabled.app_error", nil, "", http.StatusNotImplemented).Error()))
+				err := model.NewAppError("SyncLdap", "ent.ldap.disabled.app_error", nil, "", http.StatusNotImplemented)
+				mlog.Error("Not executing ldap sync because ldap is not available", mlog.Err(err))
```
Comment (hanzei) [inline app/ldap.go]: Do we really need to craft an error for here? It doesn't contain any additional information.

File: app/server.go [hunk 3]
```diff
@@ -505,7 +507,8 @@ func (s *Server) Start() error {
 	}
 	s.ListenAddr = listener.Addr().(*net.TCPAddr)
 
-	mlog.Info(fmt.Sprintf("Server is listening on %v", listener.Addr().String()))
+	logListeningPort := fmt.Sprintf("Server is listening on %v", listener.Addr().String())
+	mlog.Info(logListeningPort)
```
Comment (hanzei) [inline app/server.go]: We usually use `Starting X` and then `X stared` in our logs. What do you think about adopting this pattern and using:
```suggestion
	mlog.Info("Started Server", mlog.String("port", listener.Addr().String())
```

Comment (hanzei) [reply app/server.go]: Maybe also cut the three dots from https://github.com/mattermost/mattermost-server/pull/13370/files#diff-fb51175d7d8419ee63208a0e76554ab3R443

File: app/server.go [hunk 4]
```diff
@@ -215,8 +215,10 @@ func NewServer(options ...Option) (*Server, error) {
 		}
 	})
 
-	mlog.Info(fmt.Sprintf("Current version is %v (%v/%v/%v/%v)", model.CurrentVersion, model.BuildNumber, model.BuildDate, model.BuildHash, model.BuildHashEnterprise))
-	mlog.Info(fmt.Sprintf("Enterprise Enabled: %v", model.BuildEnterpriseReady))
+	logCurrentVersion := fmt.Sprintf("Current version is %v (%v/%v/%v/%v)", model.CurrentVersion, model.BuildNumber, model.BuildDate, model.BuildHash, model.BuildHashEnterprise)
```
Comment (hanzei) [inline app/server.go]: I would like see also see this part converted to structured logging. This would help understanding the meaning of the field. What do you think about moving this log messages into `app/server_app_adapters.go:58`. This way they would also be shown as the fist message, which is handy.

**Comment by hanzei** [comment]
Glad to see using completely moving away from structured logging :+1: 

Also: Is there an EE PR?

File: app/server.go (hunk 4 above)
Comment (jespino) [reply app/server.go]: I didn't converted it because they were informative messages during the start-up and are thought for being human readable. I would like to have others input here like @jwilander and @lieut-data

File: app/server.go (hunk 3 above)
Comment (jespino) [reply app/server.go]: I didn't converted it because they were informative messages during the start-up and are thought for being human readable.

File: app/ldap.go (hunk 2 above)
Comment (jespino) [reply app/ldap.go]: Log messages aren't translated, errors are, I was trying to keep it as less intrusive as possible

File: api4/system_test.go (hunk 1 above)
Comment (jespino) [reply api4/system_test.go]: Good point

**Comment by jespino** [comment]
@hanzei there is not a EE check yet because there are a lot of things to migrate in enterprise yet, we need to migrate it and then add the check

File: app/ldap.go (hunk 2 above)
Comment (hanzei) [reply app/ldap.go]: To me this log message is written for admins, which don't need translated error messages.

Comment (jespino) [reply app/ldap.go]: fair point, I going to remove the AppError creation there

File: app/server.go (hunk 4 above)
Comment (lieut-data) [reply app/server.go]: I can see arguments both ways -- given that we're unlikely to chart these numbers / parameters, I might lean towards leaving it human readable for now. Or, even duplicating with structured parameters, given this is a low-volume log.

File: app/email_batching.go [hunk 5]
```diff
@@ -107,7 +107,7 @@ func (job *EmailBatchingJob) CheckPendingEmails() {
 	// without actually sending emails
 	job.checkPendingNotifications(time.Now(), job.server.sendBatchedEmailNotification)
 
-	mlog.Debug(fmt.Sprintf("Email batching job ran. %v user(s) still have notifications pending.", len(job.pendingNotifications)))
+	mlog.Debug("Email batching job ran. User(s) still have notifications pending.", mlog.Int("number_of_users", len(job.pendingNotifications)))
```
Comment (lieut-data) [inline app/email_batching.go]: Since we're touching, maybe get rid of the awkward `User(s)` and just say, `Some users still have...`?

File: app/server.go (hunk 4 above)
Comment (jespino) [reply app/server.go]: I like the idea of keep both, structured logging and human readable.

Comment (jespino) [reply app/server.go]: Done

File: app/server.go (hunk 3 above)
Comment (jespino) [reply app/server.go]: Keeping here the human readable and added the structured logging at the same time.

File: app/email_batching.go (hunk 5 above)
Comment (jespino) [reply app/email_batching.go]: Done

File: app/server.go [hunk 6]
```diff
@@ -215,8 +215,17 @@ func NewServer(options ...Option) (*Server, error) {
...
+	mlog.Info(
+		logCurrentVersion,
+		mlog.String("current_version", model.CurrentVersion),
+		mlog.String("build_number", model.BuildNumber),
+		mlog.String("build_date", model.BuildDate),
+		mlog.String("build_hash", model.BuildHash),
+		mlog.String("build_hash_enterprise", model.BuildHashEnterprise),
+	)
+	logEnterpriseEnabled := fmt.Sprintf("Enterprise Enabled: %v", model.BuildEnterpriseReady)
+	mlog.Info(logEnterpriseEnabled)
```
Comment (hanzei) [inline app/server.go]: I Would like to just see us just using structured logging as it's well readable IMO, but I'm not blocking on this.

File: go.mod [hunk 7]
```diff
@@ -90,7 +90,7 @@ require (
 	golang.org/x/net v0.0.0-20191119073136-fc4aabc6c914
 	golang.org/x/sys v0.0.0-20191002091554-b397fe3ad8ed // indirect
 	golang.org/x/text v0.3.2
-	golang.org/x/tools v0.0.0-20191002161851-3769738f410b
+	golang.org/x/tools v0.0.0-20191130070609-6e064ea0cf2d
```
Comment (hanzei) [inline go.mod]: That seams like an unrelated change.

File: app/server.go (hunk 6 above)
Comment (jespino) [reply app/server.go]: I change it a bit and I think now is more clear and descriptive.

File: go.mod (hunk 7 above)
Comment (jespino) [reply go.mod]: removed, it is added again after the build

**Comment by jespino** [comment]
@hanzei PTAL

File: go.sum [hunk 8]
```diff
@@ -536,6 +536,8 @@ golang.org/x/tools v0.0.0-20190606124116-d0a3d012864b/go.mod h1:/rFqwRUd4F7ZHNgw
 golang.org/x/tools v0.0.0-20190907020128-2ca718005c18/go.mod h1:b+2E5dAYhXwXZwtnZ6UAqBI28+e2cm9otk0dWdXHAEo=
 golang.org/x/tools v0.0.0-20191002161851-3769738f410b h1:H3xgcuTEMnlg2n7BT6mWCOGuMchcPTaZ/uCC9GuV05M=
 golang.org/x/tools v0.0.0-20191002161851-3769738f410b/go.mod h1:b+2E5dAYhXwXZwtnZ6UAqBI28+e2cm9otk0dWdXHAEo=
+golang.org/x/tools v0.0.0-20191130070609-6e064ea0cf2d h1:/iIZNFGxc/a7C3yWjGcnboV+Tkc7mxr+p6fDztwoxuM=
```
Comment (hanzei) [inline go.sum]: These are also unrelated.

Comment (jespino) [reply go.sum]: Fixed
